packages/dev/core/src/Lights/areaLight.ts

```diff
@@ -0,0 +1,83 @@
+import type { Vector3 } from "../Maths/math.vector";
+import { Light } from "./light";
+import type { Effect } from "core/Materials/effect";
+import { DefaultAreaLightLTCProvider } from "core/Lights/LTC/ltcTextureTool";
+import type { IAreaLightLTCProvider } from "core/Lights/LTC/ltcTextureTool";
+import { serialize } from "../Misc/decorators";
+import type { Scene } from "core/scene";
+
+declare module "../scene" {
+    export interface Scene {
+        /**
+         * Object capable of providing LTC textures for Area Lights.
+         */
+        areaLightLTCProvider: IAreaLightLTCProvider;
+    }
+}
+
+function IsAreaLightsReady(scene: Scene): boolean {
+    if (scene.areaLightLTCProvider.ltc1Texture && scene.areaLightLTCProvider.ltc2Texture) {
+        return scene.areaLightLTCProvider.ltc1Texture.isReady() && scene.areaLightLTCProvider.ltc2Texture.isReady();
+    }
+
+    return false;
+}
+
+/**
+ * Abstract Area Light class that servers as parent for all Area Lights implementations.
+ * The light is emitted from the area in the -Z direction.
+ */
+export abstract class AreaLight extends Light {
+    protected _position: Vector3;
+
+    /**
+     * Area Light position.
+     */
+    @serialize()
+    public get position(): Vector3 {
+        return this._position;
+    }
+    /**
+     * Area Light position.
+     */
+    public set position(value: Vector3) {
+        this._position = value;
+    }

```

Comment:
Since the getter and setter are just updating the value directly, you can just do this:
```suggestion
    public position: Vector3;
```

Reply:
Actually, I was wrong, position works just fine.